Source/NSSpeechRecognizer.m

```diff
@@ -23,8 +23,241 @@
+             object: nil];
+
+      [[NSNotificationCenter defaultCenter]
+        addObserver: self
+           selector: @selector(processAppStatusNotification:)
+               name: NSApplicationDidBecomeActiveNotification
+             object: nil];
+
+      [[NSNotificationCenter defaultCenter]
+        addObserver: self
+           selector: @selector(processAppStatusNotification:)
+               name: NSApplicationDidResignActiveNotification
+             object: nil];
+
+      _delegate = nil;
+      _blocksOtherRecognizers = NO;
+      _listensInForegroundOnly = YES;
+      _uuid = [NSUUID UUID];
+      
+      if (nil == _speechRecognitionServer && !_serverLaunchTested)
+        {
+          unsigned int i = 0;
+          
+          // Wait for up to five seconds  for the server to launch, then give up.
+          for (i = 0 ; i < 50 ; i++)
+            {
+              _speechRecognitionServer = [NSConnection
+                                           rootProxyForConnectionWithRegisteredName: SPEECH_RECOGNITION_SERVER
+                                                                               host: nil];
+              RETAIN(_speechRecognitionServer);
+              if (nil != _speechRecognitionServer)
+                {
+                  NSDebugLog(@"Server found!!!");
+                  break;
+                }
+              [NSThread sleepForTimeInterval: 0.1];
+            }
+          
+          // Set a flag so we don't bother waiting for the speech recognition server to
+          // launch the next time if it didn't work this time.
+          _serverLaunchTested = YES;
+        }
+    }
+
+  [_speechRecognitionServer addClient];  // do this to update the client count;
+  
+  return self;
+}
+
+- (void) dealloc
+{
+  [[NSDistributedNotificationCenter defaultCenter] removeObserver: self];
+  [[NSNotificationCenter defaultCenter] removeObserver: self];
+  _delegate = nil;
+  [super dealloc];
+}
+
+// Delegate
+- (id<NSSpeechRecognizerDelegate>) delegate
+{
+  return _delegate;
+}
+
+- (void) setDelegate: (id<NSSpeechRecognizerDelegate>)delegate
+{
+  _delegate = delegate;
+}
+
+// Configuring...
+- (NSArray *) commands
+{
+  return _commands;
+}
+
+- (void) setCommands: (NSArray *)commands
+{
+  ASSIGNCOPY(_commands, commands);
+}
+
+- (NSString *) displayCommandsTitle
+{
+  return _displayCommandsTitle;
+}
+
+- (void) setDisplayCommandsTitle: (NSString *)displayCommandsTitle
+{
+  ASSIGNCOPY(_displayCommandsTitle, displayCommandsTitle);
+}
+
+- (BOOL) listensInForegroundOnly
+{
+  return _listensInForegroundOnly;
+}
+
+- (void) setListensInForegroundOnly: (BOOL)listensInForegroundOnly
+{
+  _listensInForegroundOnly = listensInForegroundOnly;
+}
+
+- (BOOL) blocksOtherRecognizers
+{
+  return _blocksOtherRecognizers;
+}
+
+- (void) setBlocksOtherRecognizers: (BOOL)blocksOtherRecognizers
+{
+  if (blocksOtherRecognizers == YES)
+    {
+      [_speechRecognitionServer addToBlockingRecognizers: [_uuid UUIDString]];
```

Comment:
Instead of overriding performSelector: and performSelector:withObject: I have simply wrapped everywhere where the server is called with exception/error handling logic.   There are only two places in which this is done in the recognizer so I didn't see the point in centralizing the error handling.